pyccel/codegen/pipeline.py

```diff
@@ -196,12 +212,65 @@ def handle_error(stage):
 
         #------------------------------------------------------
         # TODO: collect dependencies and proceed recursively
-    #    if recursive:
-    #        for dep in parser.sons:
-    #            # Call same function on 'dep'
-    #            pass
+        # if recursive:
+        #     for dep in parser.sons:
+        #         # Call same function on 'dep'
+        #         pass
         #------------------------------------------------------
 
+        # Iterate over the internal_libs list and determine if the printer
+        # requires an internal lib to be included.
+        for lib in internal_libs:
+            if lib in codegen.get_printer_imports():
+                # get the include folder path and library files
+                if lib not in internal_libs_name:
+                    # get the library folder name
+                    lib_name = internal_libs[lib]
+                    # get path to pyccel/stdlib/lib_name
+                    import pyccel.stdlib as stdlib_folder
+                    stdlib_path = os.path.dirname(stdlib_folder.__file__)
+                    lib_path = os.path.join(stdlib_path, lib_name)
+                    # remove library folder to avoid missing files and copy
+                    # new one from pyccel stdlib
+                    lib_dest_path = os.path.join(pyccel_dirpath, lib_name)
+                    if os.path.exists(lib_dest_path):
+                        shutil.rmtree(lib_dest_path)
+                    shutil.copytree(lib_path, lib_dest_path)
+
+                    # stop after copying lib to __pyccel__ directory for
+                    # convert only
+                    if convert_only:
+                        continue
+
+                    # get library source files
+                    source_files = []
+                    for e in os.listdir(lib_dest_path):
+                        if e.endswith(lang_ext_dict[language]):
+                            source_files.append(os.path.join(lib_dest_path, e))
+
+                    # compile library source files
+                    flags = construct_flags(f90exec,
+                                            fflags=fflags,
+                                            debug=debug,
+                                            includes=[lib_dest_path])
```

Comment:
Do we really need to recalculate the flags after each loop?

Reply:
yes, the includes are different on each iteration.

Reply:
This feels like overkill just for the includes. But I don't think it makes sense to fix this before fixing #517